templates/flavors/default/patches/control-plane.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureCluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  networkSpec:
+    apiServerLB:
+      # We pre-create this public IP and the DNS name to use it in the
+      # worker node's /etc/hosts.
+      frontendIPs:
+        - name: ${CLUSTER_NAME}-api-lb
+          publicIP:
+            name: ${CLUSTER_NAME}-api-lb
+            dnsName: ${CLUSTER_NAME}-${APISERVER_LB_DNS_SUFFIX}.${AZURE_LOCATION}.cloudapp.azure.com
```

Comment:
Could we incorporate this into the base template for all the other self-managed templates that aren't derived from the `default` flavor?

Reply:
Waiting on a resolution here https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5209#discussion_r1821104012, and then will pull this change into `base`.

Reply:
We can probably land these changes to *all* templates in a phased approach. Landing a working pattern of this ASAP will allow folks managing their own templates to integrate with this new solution sooner.

wdyt @nojnhuh?

Reply:
sgtm

Reply:
Actually, why don't I follow it up with another PR and unblock the team asap with landing this one ?